src/TwigComponent/src/MountedComponent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+final class MountedComponent
+{
+    public function __construct(
+        private object $component,
+        private ComponentAttributes $attributes,
+        private ComponentMetadata $metadata
```

Comment:
It's minor, but for some reason, it's bothering me that `ComponentMetadata` is inside this object. I think it's because the `$component` and `$attributes` are both "run-time" values (i.e. the user can pass in different values) but `ComponentMetadata` is static config.

And, looking at where we use it, I think it may not be needed. If we:

* Add `string $name` to `MountedComponent`
* In `ComponentRenderer`, inject `ComponentFactory` and use it to get the metadata - e.g. `$this->factory->metadataFor($mountedComponent->getName())`. THEN, ONLY pass the `->getTemplate()` into `PreRenderEvent`... because that's all we need! We currently use `->getName()` via `AddLiveAttributesSubscriber` calling `LiveComponentRuntime::getLiveAttributes`, but that can be replaced with `$mountedComponent->getName()`/

Reply:
Sounds good, I'll play with this idea.